lib/pure/endians.nim

```diff
@@ -98,16 +124,734 @@ else:
+
+  proc bigEndian64*(outp, inp: pointer) {.inline, deprecated:
+      "Deprecated since v1.1.0; use `fromBytesXX/toBytesXX` instead".} =
+    copyMem(outp, inp, 8)
+
+  proc bigEndian32*(outp, inp: pointer) {.inline, deprecated:
+      "Deprecated since v1.1.0; use `fromBytesXX/toBytesXX` instead".} =
+    copyMem(outp, inp, 4)
+
+  proc bigEndian16*(outp, inp: pointer) {.inline, deprecated:
+      "Deprecated since v1.1.0; use `fromBytesXX/toBytesXX` instead".} =
+    copyMem(outp, inp, 2)
+else:
+  proc littleEndian64*(outp, inp: pointer) {.inline, deprecated:
+      "Deprecated since v1.1.0; use `fromBytesXX/toBytesXX` instead".} =
+    copyMem(outp, inp, 8)
+
+  proc littleEndian32*(outp, inp: pointer) {.inline, deprecated:
+      "Deprecated since v1.1.0; use `fromBytesXX/toBytesXX` instead".} =
+    copyMem(outp, inp, 4)
+
+  proc littleEndian16*(outp, inp: pointer){.inline, deprecated:
+      "Deprecated since v1.1.0; use `fromBytesXX/toBytesXX` instead".} =
+    copyMem(outp, inp, 2)
+
+  proc bigEndian64*(outp, inp: pointer) {.inline, deprecated:
+      "Deprecated since v1.1.0; use `fromBytesXX/toBytesXX` instead".} =
+    swapEndian64(outp, inp)
+
+  proc bigEndian32*(outp, inp: pointer) {.inline, deprecated:
+      "Deprecated since v1.1.0; use `fromBytesXX/toBytesXX` instead".} =
+    swapEndian32(outp, inp)
+
+  proc bigEndian16*(outp, inp: pointer) {.inline, deprecated:
+      "Deprecated since v1.1.0; use `fromBytesXX/toBytesXX` instead".} =
+    swapEndian16(outp, inp)
+
+
+# New API starts here
+
+func slowSwap16(a: uint16): uint16 {.inline.} =
+  result = (a shl 8) or (a shr 8)
+
+func slowSwap32(a: uint32): uint32 {.inline.} =
+  var a = ((a shl  8) and 0xff00ff00'u32) or
+          ((a shr  8) and 0x00ff00ff'u32)
+  result = (a shl 16) or (a shr 16)
+
+func slowSwap64(a: uint64): uint64 {.inline.} =
+  var a = ((a shl 8) and 0xff00ff00ff00ff00'u64) or
+          ((a shr 8) and 0x00ff00ff00ff00ff'u64)
+  a = ((a shl 16) and 0xffff0000ffff0000'u64) or
+      ((a shr 16) and 0x0000ffff0000ffff'u64)
+  result = (a shl 32) or (a shr 32)
+
+when useBuiltinSwap:
+  template swap16(a: uint16): uint16 = builtinSwap16(a)
+  template swap32(a: uint32): uint32 = builtinSwap32(a)
+  template swap64(a: uint64): uint64 = builtinSwap64(a)
 else:
-  proc littleEndian64*(outp, inp: pointer) {.inline.} = copyMem(outp, inp, 8)
-  proc littleEndian32*(outp, inp: pointer) {.inline.} = copyMem(outp, inp, 4)
-  proc littleEndian16*(outp, inp: pointer){.inline.} = copyMem(outp, inp, 2)
-  proc bigEndian64*(outp, inp: pointer) {.inline.} = swapEndian64(outp, inp)
-  proc bigEndian32*(outp, inp: pointer) {.inline.} = swapEndian32(outp, inp)
-  proc bigEndian16*(outp, inp: pointer) {.inline.} = swapEndian16(outp, inp)
+  template swap16(a: uint16): uint16 = slowSwap16(a)
+  template swap32(a: uint32): uint32 = slowSwap32(a)
+  template swap64(a: uint64): uint64 = slowSwap64(a)
+
+
+# TODO had to remove {.since: (1,1).} because it broke `koch runCI`
```

Comment:
Well, .since is not negotiable here. How come "koch runCI" failed with this? We use it successfully everywhere else.